contracts/registry/src/trust_score.rs

```
@@ -123,6 +123,41 @@ pub struct TrustHistory {
+pub fn pair_discount_scale(k: u32) -> u64 {
+    if k <= 1 {
+        SCALE
+    } else {
+        let k64 = k as u64;
+        SCALE / (k64 * k64)
+    }
+}
+
+/// Calculates the global counterparty diversity factor (scale 2^32) based on number of unique counterparties.
+pub fn diversity_factor(unique_count: u32) -> u64 {
+    if unique_count >= 1 {
```

Clippy (1.97.1)

[warning] this `if` has identical blocks

(warning)